### ISSUE TITLE
Bug: proxy inside wp filters instead of localhost:3000

### ISSUE THREAD

**Issue by SergiArias** [roots/sage#2928]
### Terms

- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This request is not a duplicate of an existing issue
- [X] I have read the [docs](https://roots.io/docs/) and followed them (if applicable)
- [X] I have seached the [Roots Discourse](https://discourse.roots.io/) for answers and followed them (if applicable)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Description

Maybe it's not directly a sage bug, but it affects the natural intended workflow of bud.

When I filter the menu to use img instead of text, the asset()->uri gives me the proxy instead of the localhost:3000 address and this forces me to “unsafely” access the url to see the images. I don’t quite understand why this happens as outside of the filter it gives the localhost:3000 address.

Can it be uri sanitization inside the navwalker?

### Steps To Reproduce

```php
add_filter( 'wp_nav_menu_items', function ( $items ) {
    $asset = asset('assets/images/profile.svg')->uri();
    echo $asset; //<- this returns http://localhost:3000/

    $items = str_replace(
        '{{profile}}',
        '<img class="h-7 w-auto ls:hidden" src="' . $asset . '" alt="">',
        $items
    ); //<- this returns https://proxy.test/

    return $items;
});
```

### Expected Behavior

Always return the localhost address while in dev

### Actual Behavior

It returns the proxy address instead of the localhost one.

### Versions

10.beta-2

**Comment by retlehs** [comment]
@SergiArias i'm unable to reproduce this bug

```php
$asset = asset('assets/images/example.svg')->uri();
```

results in:

```
http://example.test/app/themes/sage/public/assets/images/example.8f7dad.svg
```

**Comment by SergiArias** [comment]
But this is when you run `yarn build`, isn't? I was referring to the development environment. The expected behaviour when running `yarn dev` is that the assets also go through localhost:3000, this way the browser won't ask for permission to load them due to lack of certificate.

I am using Laravel Valet by the way.

**Comment by retlehs** [comment]
unable to reproduce. from `yarn dev`, the result is:

```
http://localhost:3000/app/themes/sage/public/assets/images/example.svg
```

**Comment by SergiArias** [comment]
Yes, the result is correct for most cases, but not inside filters. Please, try the following: Create a menu with a name `{{profile}}` and use the following filter:

```php
add_filter( 'wp_nav_menu_items', function ( $items ) {
    $asset = asset('assets/images/profile.svg')->uri();
    echo $asset; //<- this returns http://localhost:3000/

    $items = str_replace(
        '{{profile}}',
        '<img class="h-7 w-auto ls:hidden" src="' . $asset . '" alt="">',
        $items
    ); //<- this returns https://proxy.test/

    return $items;
});
```

**Comment by retlehs** [comment]
i'm not able to reproduce within a filter. using the `asset()` function within the provided filter in `app/filters.php` is having the expected results.

from the OP:

> Maybe it's not directly a sage bug

maybe bring some more evidence of a bug that's in sage? i'm spending time here trying to debug an issue that doesn't appear to actually be an issue

**Comment by kellymears** [comment]
@SergiArias what is the html output of that filter (the source as it is in the browser). The proxy middleware should replace any `src` attribute that contains the public path. Please use `view-source` or the body of the response from the network tab of devtools. Browsers do a lot to normalize html, we want to see the raw response, because that's what the proxy middleware is working with.

**Comment by SergiArias** [comment]
Of course @kellymears this is the source (only the menu part where the filter is applied):

```html
<nav class="nav-ecommerce ls:ml-8" aria-label="eCommerce">
  <ul id="menu-ecommerce" class="nav flex space-x-[1.125rem] mr-[1.125rem] ls:space-x-7 ls:mr-7">
    <li class="menu-item menu-profile"><a href="http://localhost:3000/mi-cuenta/"><img class="h-7 w-auto ls:hidden" src="https://covidtest21.test/wp-content/themes/sage-covidtest21/public/assets/images/profile.svg" alt=""><img class="hidden h-8 ls:block" src="https://covidtest21.test/wp-content/themes/sage-covidtest21/public/assets/images/profile_ls.svg" alt=""></a></li>
    <li class="menu-item menu-cart"><a href="http://localhost:3000/carrito/"><img class="h-7 w-auto ls:hidden" src="https://covidtest21.test/wp-content/themes/sage-covidtest21/public/assets/images/cart.svg" alt=""><img class="hidden h-8 ls:block" src="https://covidtest21.test/wp-content/themes/sage-covidtest21/public/assets/images/cart_ls.svg" alt=""></a></li>
  </ul>
</nav>
```

The body of the response in network tools tab gives me the same result.

Anyway, if it's not reproducible by you maybe it's better to wait for other users to confirm they have the same problem before spending too much time debugging. It's not a big deal of a problem.

**Comment by retlehs** [comment]
i think this might be fixed as of bud v5.2.0
